plugin/storage/cassandra/dependencystore/storage.go

```diff
@@ -26,60 +28,96 @@ import (
 	casMetrics "github.com/jaegertracing/jaeger/pkg/cassandra/metrics"
 )
 
+// IndexMode determines how the dependency data is indexed.
+type IndexMode int
+
 const (
-	depsInsertStmt = "INSERT INTO dependencies(ts, ts_index, dependencies) VALUES (?, ?, ?)"
-	depsSelectStmt = "SELECT ts, dependencies FROM dependencies WHERE ts_index >= ? AND ts_index < ?"
+	// SASIEnabled is used when the dependency table is SASI indexed.
+	SASIEnabled IndexMode = iota
+
+	// SASIDisabled is used when the dependency table is NOT SASI indexed.
+	SASIDisabled
+
+	depsInsertStmtSASI = "INSERT INTO dependencies(ts, ts_index, dependencies) VALUES (?, ?, ?)"
+	depsInsertStmt     = "INSERT INTO dependenciesv2(ts, date_bucket, dependencies) VALUES (?, ?, ?)"
+	depsSelectStmtSASI = "SELECT ts, dependencies FROM dependencies WHERE ts_index >= ? AND ts_index < ?"
+	depsSelectFmt      = "SELECT ts, dependencies FROM dependenciesv2 WHERE date_bucket IN (%s) AND ts >= ? AND ts < ?"
+	dateFmt            = "20060102"
+	day                = 24 * time.Hour
 )
 
 // DependencyStore handles all queries and insertions to Cassandra dependencies
 type DependencyStore struct {
 	session                  cassandra.Session
 	dependenciesTableMetrics *casMetrics.Table
 	logger                   *zap.Logger
+	indexMode                IndexMode
 }
 
 // NewDependencyStore returns a DependencyStore
 func NewDependencyStore(
 	session cassandra.Session,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
+	indexMode IndexMode,
```

Comment:
Is there anything preventing someone from passing in an undefined `IndexMode`, like IndexMode(10) for e.g.?

Reply:
ill validate here